### DISCUSSION TITLE
How to upload file to 'local' and 's3' storage on the same time.

### DISCUSSION THREAD

**Discussion by pashtostories** [spatie/laravel-medialibrary#2292] [Q&A] [answered]
The reason for uploading the same file locally and to s3 bucket in the same time is to be able in the future to switch fetching and serving files from any of these storages easily and without any hassles. Like if for some reason I don't want to serve the files from s3 bucket for any reason, then I want to be able to edit my config and switch local to s3. I would love to know best practices to achieve this functionality, because I want to be risk free in term of serving files from my app. Thank you.

**Comment by drbyte** [comment, marked as answer]
I'm not sure if this is the best answer, but it's one approach: 
Once the media is saved to one location, `copy` it to another location as described in https://spatie.be/docs/laravel-medialibrary/v9/advanced-usage/moving-media#copying-media
